ui/src/shared/apis/flux/query.ts

```diff
@@ -73,10 +74,14 @@ export const executeQuery = async (
   const reject = () => {
     let bodyError
 
-    try {
-      bodyError = JSON.parse(xhr.responseText).error
-    } catch {
-      bodyError = null
+    if (xhr.status === 404 && xhr.responseText === FLUX_NOT_ENABLED_RESPONSE) {
+      bodyError = xhr.responseText
```

Comment:
i think we might be able to keep this logic a little more generic, rather than specially handling that particular error message here

```typescript
let bodyError

try {
  bodyError = JSON.parse(xhr.responseText).error
} catch {
  if (xhr.responseText && xhr.responseText.trim() !== '') {
    bodyError = xhr.responseText
  }
}

const statusError = xhr.statusText
// ...
```

Reply:
Oh nice! I like that a lot more